docs/admin/templates/extending-templates/prebuilt-workspaces.md

```diff
@@ -0,0 +1,199 @@
+# Prebuilt workspaces
+
+## Overview
+
+Prebuilt workspaces let you pre-provision and maintain a pool of ready-to-deploy workspaces.
+Instead of creating a new workspace when a developer requests one, if a workspace matches a preset defined in the
+template parameters, Coder assigns an existing instance, reducing setup time significantly.
+
+## Prerequisites
+
+- Premium license
+- Use `coder/coder` Terraform provider `>= 2.3.0-pre2` in your template (**TODO: update with latest version**)
+- Enable the `workspace-prebuilds` [experiment](../../../reference/cli/server.md#--experiments)
+
+## Configuration
+
+1. In your Terraform template, add a `prebuilds` block within a `coder_workspace_preset` block:
+
+   ```hcl
+   data "coder_workspace_preset" "goland" {
+     name = "GoLand: Large"
+     parameters = {
+       jetbrains_ide = "GO"
+       cpus          = 8
+       memory        = 16
+     }
+     prebuilds {
+       instances = 3
+     }
+   }
+   ```
+
+   Do not define values for `coder_workspace` or `coder_workspace_owner`.
+   These values are [replaced](#resource-replacement) when the workspace is deployed, and if Coder detects an existing
+   value, it will destroy the prebuilt workspace and create a new one.
```

Comment:
@dannykopping flagging this for review - I want to bring a notice about it closer to where it might happen, but I'm not sure if I captured the quick version correctly.

I
```suggestion
   Do not define values for `coder_workspace` or `coder_workspace_owner`.
   These values are [replaced](#resource-replacement) when the workspace is deployed, and if Coder detects an existing
   value, it will destroy the prebuilt workspace and create a new one for the user.
```

Reply:
These values cannot be set, they're derived.
I'll revert this.